[rum] addTiming docs (#9606)
DataDog/documentation: content/en/real_user_monitoring/browser/monitoring_page_performance.md

The RUM SDK automatically monitors frameworks that rely on hash (`#`) navigation. The SDK watches for `HashChangeEvent` and issues a new view. Events coming from an HTML anchor tag which do not affect the current view context are ignored.

## Add your own performance timing
On top of RUM's default performance timing, you may measure where your application is spending its time with greater flexibility. The `addTiming` API provides you with a simple way to add extra performance timing. For example, you can add a timing when your hero image has appeared:

```html
<html>
  <body>
    <img onload="DD_RUM.addTiming('hero_image')" src="/path/to/img.png" />
  </body>
</html>
```

Or when users first scroll:

```javascript
document.addEventListener("scroll", function handler(e) {
    e.currentTarget.removeEventListener(e.type, handler); //Remove the event listener so that it only triggers once
    DD_RUM.addTiming('first_scroll');

> });

Reviewer: In order to be more readable, if we don't care about IE support for the example:
```js
document.addEventListener("scroll", function () {
    DD_RUM.addTiming('first_scroll');
}, { once: true });
```
otherwise:
```js
document.addEventListener("scroll", function handler() {
    // Remove the event listener so that it only triggers once
    document.removeEventListener("scroll", handler); 
    DD_RUM.addTiming('first_scroll');
});
```

Author: Let's go for the IE support as it's not much longer, I modified the comment position

Reviewer: any reason to remove the listener from the event target instead of reusing document?

Author: If we remove it from the document we can potentially delete all event listeners attached to scroll right?

Reviewer: No, only the one with the same handler reference.
```
document.removeEventListener("scroll", handler); 
```

```

> 

Reviewer: Should we add something about facet creation?
like RUM Explorer > Add a facet > @view.custom_timings.first_scroll

The RUM SDK automatically monitors frameworks that rely on hash (`#`) navigation. The SDK watches for `HashChangeEvent` and issues a new view. Events coming from an HTML anchor tag which do not affect the current view context are ignored.

## Add your own performance timing
On top of RUM's default performance timing, you may measure where your application is spending its time with greater flexibility. The `addTiming` API provides you with a simple way to add extra performance timing. For example, you can add a timing when your hero image has appeared:

```html
<html>
  <body>
    <img onload="DD_RUM.addTiming('hero_image')" src="/path/to/img.png" />
  </body>
</html>
```

Or when users first scroll:

```javascript

> document.addEventListener("scroll", function handler(e) {

Reviewer: ```suggestion
document.addEventListener("scroll", function handler() {
```

    //Remove the event listener so that it only triggers once

>     e.currentTarget.removeEventListener(e.type, handler);

Reviewer: ```suggestion
    document.removeEventListener("scroll", handler);
```

    DD_RUM.addTiming('first_scroll');
});
```

> Once the timing is sent, you must [create a measure][15] before graphing it in RUM analytics or in dashboards.

Reviewer: ```suggestion
Once the timing is sent, you must [create a measure][15] before graphing it in RUM analytics or in dashboards. Your performance timings will stored under `@view.custom_timings.[YOUR_TIMING_NAME]`.
```
IMO something like that could be valuable for customers to find where it is.
wdyt?

> Once the timing is sent, the timing will be accessible as `@view.custom_timings.<timing_name>` (e.g. `@view.custom_timings.first_scroll`). You must [create a measure][15] before graphing it in RUM analytics or in dashboards.

Reviewer 2: ```suggestion
Once the timing is sent, the timing will be accessible as `@view.custom_timings.<timing_name>` (For example, `@view.custom_timings.first_scroll`). You must [create a measure][15] before graphing it in RUM analytics or in dashboards.
```
